.github/workflows/rust.yml

```diff
@@ -30,6 +30,21 @@ jobs:
     - run: rustup update
     - run: cargo fmt --all --check
 
+  cargo-deny:
```

Comment:
Should we run it every time on PR? Maybe we can skip running it if `Cargo.lock` was unchanged?

Reply:
It's cheap to run, so I don't think it matters much. We run it on much larger repos on each commit.

Reply:
That makes sense it also detects new vulnerabilities which is very useful: should we also update our dependencies highlighted by cargo-deny?